tensorboard/webapp/widgets/custom_modal/custom_modal_component.ts

```diff
@@ -43,51 +46,51 @@ export interface ModalContent {
         position: fixed;
         left: 0;
         z-index: 9001;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
       }
 
       .content {
         position: absolute;
+        pointer-events: initial;
       }
     `,
   ],
 })
-export class CustomModalComponent implements OnInit {
+export class CustomModalComponent implements OnInit, OnDestroy {
   @Output() onOpen = new EventEmitter<void>();
   @Output() onClose = new EventEmitter<void>();
 
   readonly visible$ = new BehaviorSubject(false);
   private canClose = true;
 
-  @ViewChild('content', {static: false})
+  @ViewChild('content', {static: true})
   private readonly content!: ElementRef;
 
-  private clickListener: () => void = this.maybeClose.bind(this);
+  static latestInstance: CustomModalComponent;
+  parentEmbeddedViewRef?: EmbeddedViewRef<unknown>;
+  clickListener: () => void = this.maybeClose.bind(this);
 
-  constructor(private readonly viewRef: ViewContainerRef) {}
+  constructor() {
+    CustomModalComponent.latestInstance = this;
+  }
 
   ngOnInit() {
-    this.visible$.subscribe((visible) => {
-      // Wait until after the next browser frame.
+    this.visible$.pipe(filter((visible) => !!visible)).subscribe((visible) => {
```

Comment:
Making onClose emission dependent on visible$ can cause surprising behavior: e.g. visible$ will be false on init, which triggers onClose on modal init, which leads to erratic unit tests.

Changed to simply emit onClose separately in the close() method.